pkg/client/types.go

```diff
@@ -34,11 +34,20 @@ const (
 // Error is an error returned by the API.
 type Error struct {
 	Type ErrorType
-	Msg  string
+	ErrorMsg  string
+	Query string
+	StatusCode int
 }
 
 func (e *Error) Error() string {
```

Comment:
Do you have an example of what this log would look like?

Reply:
`
bad_response: unable to fetch node CPU metrics: {"status":"error","errorType":"bad_data","error":"unknown position: parse error: no expression found in input"} for status code: 400, query: query=sum%28max+by+%28container_name%2C+pod_name%29+%28rate%28container_cpu_user_seconds_total%7Bnamespace%3D%22ngrok-controlplane-blue%22%2Cpod_name%3D~%22creds-global-c55fbf6b4-fsb8b%7Ccreds-global-c55fbf6b4-hlvtc%7Ccreds-global-c55fbf6b4-qbsfw%22%2C+container_name%21%3D%27%27%7D%5B5m%5D%29+%2B+rate%28container_cpu_system_seconds_total%7Bnamespace%3D%22ngrok-controlplane-blue%22%2Cpod_name%3D~%22creds-global-c55fbf6b4-fsb8b%7Ccreds-global-c55fbf6b4-hlvtc%7Ccreds-global-c55fbf6b4-qbsfw%22%2C+container_name%21%3D%27%27%7D%5B5m%5D%29%29%29+by+%28pod_name%2Ccontainer_name%29&time=1732224599.633
`
This is what the error would look like, except in this case I changed the query being sent to be `query=invalid_query=!!!`